Assets/_BForBoss/_Weapons/Scripts/AutomaticWeaponBehaviour.cs

```diff
@@ -22,14 +22,13 @@ protected override void OnFireInputAction(InputAction.CallbackContext context)
 
         protected override void Update()
         {
-            if (!_isFiring)
+            if (_isFiring)
             {
-                return;
+                _weapon.DecrementElapsedTimeRateOfFire(Time.deltaTime);
```

Comment:
Shouldn't this also decrement even if not firing?

Reply:
yup you are correct, updating